src/common/selftest/selftest_firstlayer_interface.hpp

```diff
@@ -0,0 +1,27 @@
+/**
+ * @file selftest_firstlayer_interface.hpp
+ * @brief interface between selftest and first layer calibration
+ */
+#pragma once
+#include <cstdint>
+#include "selftest_firstlayer_config.hpp"
+
+class IPartHandler;
+
+namespace selftest {
+
+/**
+ * @brief this function is cyclically called in selftest
+ * 1. handles dynamical allocation of selftest part state machine
+ * 2. calls loop and notifies other threads is state was changed
+ * 3. stores result to eeprom
+ * 4. handles deallocation
+ *
+ * @param pFirstLayer reference to pointer of state machine
+ * @param config      configuration structure
+ * @return true       not finished
+ * @return false      finished
```

Comment:
If documenting return values use retval instead
```
 * @retval true       not finished
 * @retval false      finished
```

Reply:
it was automatically generated, but will do as you wish